plugins/metrics/pkg/gateway/services/rbac.go

```diff
@@ -0,0 +1,257 @@
+package services
```

Comment:
In general I would discourage using `system.NewLock` here since these are meant to be long lived locks that are intended to live for the duration of the process -- the only way such a lock is guaranteed to unlock is if the client holding it is unreachable, which is probably not desireable for transactions (or unless the client making requests for the lock only lives for the duration of the API call, that would be valid, but not possible if using the `system.NewLock` abstraction)

Also, in particular the etcd implementation in `topic/scaling` of the etcd Lock will not lock between the same host as it reuses the underlying client connection, so I don't think this would achieve the effect desired here. Somewhat ironically, adding a sync.Mutex would fix that

I would suggest using revisions and embedding them in the interface, since the underlying implementation is the KV store anyways